File: app/src/main/java/com/tkdev/ironmanstash/infinity_stones/fragments/StarkFragment.java
package com.tkdev.ironmanstash.infinity_stones.fragments;


import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.animation.ObjectAnimator;
import android.graphics.Color;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.LinearInterpolator;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ScrollView;
import android.widget.Toast;

import com.tkdev.ironmanstash.R;
import com.tkdev.ironmanstash.infinity_stones.fragments.allstones.InfinityFragment;

/**
 * A simple {@link Fragment} subclass.
 */
public class StarkFragment extends Fragment {


    private ScrollView scrollView;
    private ImageView helmetView;
    private Button agreeButton;
    private MediaPlayer mediaPlayer;

    public StarkFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_stark, container, false);


        helmetView = rootView.findViewById(R.id.imageView);
        scrollView = rootView.findViewById(R.id.scroll_view);
        agreeButton = rootView.findViewById(R.id.agree_button);

        scrollView.setAlpha(0.0f);
        scrollView.setBackgroundColor(getResources().getColor(R.color.scrollBackground));
        agreeButton.setAlpha(0.0f);
        helmetView.setImageResource(R.drawable.ironman_helmet_r);
        helmetView.setAlpha(0.2f);


        return rootView;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        backgroundSong();

        helmetView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                helmetView.setOnClickListener(null);
                getView().setBackgroundColor(0xFF222222);
                helmetView.animate().alpha(1.0f).setDuration(1500);
                scrollView.animate().alpha(1.0f).setDuration(3000).setListener(new AnimatorListenerAdapter() {
                    @Override
                    public void onAnimationEnd(Animator animation) {
                        super.onAnimationEnd(animation);
                        ObjectAnimator scrollAnimator = ObjectAnimator.ofInt(scrollView, "scrollY", scrollView.getChildAt(0).getHeight() - scrollView.getHeight());
                        scrollAnimator.setDuration(30000);
                        scrollAnimator.setInterpolator(new LinearInterpolator());
                        scrollAnimator.start();
                        scrollAnimator.addListener(new AnimatorListenerAdapter() {


                            @Override
                            public void onAnimationEnd(Animator animation) {
                                agreeButton.animate().alpha(1.0f).setDuration(5000);
                            }
                        });
                    }
                });

            }
        });


        agreeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getActivity().getSupportFragmentManager()
                        .beginTransaction()
                        .replace(R.id.activity_container, new InfinityFragment())
                        .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                        .commit();
                if (mediaPlayer != null) {
                    mediaPlayer.release();
                }
                mediaPlayer = null;
            }
        });
    }

    private void backgroundSong() {
        mediaPlayer = MediaPlayer.create(getContext(), R.raw.endgame);
        mediaPlayer.seekTo(119700);
        mediaPlayer.start();

    }

    @Override
    public void onResume() {
        super.onResume();
        if (mediaPlayer != null) {
            mediaPlayer.start();
        }

    }

    @Override
    public void onStop() {
        super.onStop();
        if (mediaPlayer != null) {
            mediaPlayer.pause();
        }
    }

    @Override
    public void onPause() {
        super.onPause();
        if (mediaPlayer != null) {
            mediaPlayer.pause();
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (mediaPlayer != null) {
            mediaPlayer.release();
        }
        mediaPlayer = null;

    }
}

File: app/src/main/java/com/tkdev/ironmanstash/infinity_stones/fragments/StarkFragment.kt
package com.tkdev.ironmanstash.infinity_stones.fragments

import InfinityFragment
import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.animation.ObjectAnimator
import android.media.MediaPlayer
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.LinearInterpolator
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import com.tkdev.ironmanstash.R
import kotlinx.android.synthetic.main.fragment_stark.*

class StarkFragment : Fragment() {
    private var mediaPlayer: MediaPlayer? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_stark, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        scrollView.alpha = 0.0f
        scrollView.setBackgroundColor(resources.getColor(R.color.scrollBackground))
        agreeButton.alpha = 0.0f
        helmetView.setImageResource(R.drawable.ironman_helmet_r)
        helmetView.alpha = 0.2f
        helmetView.setOnClickListener {
            helmetView.setOnClickListener(null)
            stark_container.setBackgroundColor(-0xddddde)
            helmetView.animate().alpha(1.0f).duration = 1500
            scrollView.animate().alpha(1.0f).setDuration(3000).setListener(object : AnimatorListenerAdapter() {
                override fun onAnimationEnd(animation: Animator) {
                    super.onAnimationEnd(animation)
                    val scrollAnimator = ObjectAnimator.ofInt(scrollView, "scrollY", scrollView.getChildAt(0).height - scrollView.height)
                    scrollAnimator.duration = 30000
                    scrollAnimator.interpolator = LinearInterpolator()
                    scrollAnimator.start()
                    scrollAnimator.addListener(object : AnimatorListenerAdapter() {
                        override fun onAnimationEnd(animation: Animator) {
                            agreeButton.animate().alpha(1.0f).duration = 5000
                        }
                    })
                }
            })
        }
        agreeButton.setOnClickListener {
            parentFragmentManager
                    .beginTransaction()
                    .replace(R.id.activity_container, InfinityFragment())
                    .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                    .commit()
            mediaPlayer?.release()
        }
    }

    private fun backgroundSong() {
        mediaPlayer = MediaPlayer.create(context, R.raw.endgame)
        mediaPlayer?.seekTo(119700)
        mediaPlayer?.start()
    }

    override fun onResume() {
        super.onResume()
//       Runnable {  backgroundSong() }
    }

    override fun onStop() {
        super.onStop()
        mediaPlayer?.pause()
    }

    override fun onPause() {
        super.onPause()
        mediaPlayer?.pause()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mediaPlayer?.release()
    }
}